contracts/gateway/src/contract.rs

```diff
@@ -61,40 +50,37 @@ pub fn execute(
 
 pub mod execute {
 
-    use cosmwasm_std::{to_binary, QueryRequest, StdError, WasmMsg, WasmQuery};
+    use connection_router::state::CrossChainId;
+    use cosmwasm_std::{to_binary, QueryRequest, WasmMsg, WasmQuery};
 
     use crate::{events::GatewayEvent, state::OUTGOING_MESSAGES};
 
     use super::*;
 
-    fn contains_duplicates(msgs: &mut Vec<Message>) -> bool {
+    fn contains_duplicates(msgs: &mut Vec<NewMessage>) -> bool {
         let orig_len = msgs.len();
-        msgs.sort_unstable_by_key(|a| a.id.to_string());
-        msgs.dedup_by(|a, b| a.id == b.id);
+        msgs.sort_unstable_by_key(|a| a.cc_id.to_string());
+        msgs.dedup_by(|a, b| a.cc_id == b.cc_id);
         orig_len != msgs.len()
     }
 
     fn partition_by_verified(
         deps: DepsMut,
-        msgs: Vec<Message>,
-    ) -> Result<(Vec<Message>, Vec<Message>), ContractError> {
+        msgs: Vec<NewMessage>,
+    ) -> Result<(Vec<NewMessage>, Vec<NewMessage>), ContractError> {
         let verifier = CONFIG.load(deps.storage)?.verifier;
 
         let query_msg = aggregate_verifier::msg::QueryMsg::IsVerified {
-            messages: msgs
-                .iter()
-                .map(|m| m.clone().try_into())
-                .collect::<Result<_, _>>()
-                .map_err(|_| StdError::generic_err("invalid messages"))?, //todo: error mapping needs to get removed when gateway is mirated to NewMessage
+            messages: msgs.clone(),
         };
-        let query_response: Vec<(String, bool)> =
+        let query_response: Vec<(CrossChainId, bool)> =
             deps.querier.query(&QueryRequest::Wasm(WasmQuery::Smart {
                 contract_addr: verifier.to_string(),
                 msg: to_binary(&query_msg)?,
             }))?;
 
         Ok(msgs.into_iter().partition(|m| -> bool {
-            match query_response.iter().find(|r| m.id.to_string() == r.0) {
+            match query_response.iter().find(|r| m.cc_id == r.0) {
```

Comment:
I'm never a fan of these one-letter variable names.

Reply:
I actually don't mind really short variable names in lambdas like this

Reply:
I do prefer `msg` over `m`, i.e. if there is a good short abbreviation I'd rather use this than just a single letter, but it is acceptable to me as long as the expression is trivial

Reply:
I'd much prefer `msg` over `m`, and `res` over `r`.